### PR TITLE
Implement image_to_pdf for cli tesseract

### PR DESCRIPTION
**Pull request by mskrip**

### CODE REVIEW

File: src/pyocr/tesseract.py [hunk 1]
```diff
@@ -376,6 +376,40 @@ def image_to_string(image, lang=None, builder=None):
                              " last name tried: %s" % output_file_name)
 
 
+def image_to_pdf(image, lang=None):
```
Comment (lubo) [inline src/pyocr/tesseract.py]: libtesseract counterpart takes two extra arguments, `input_file` and `textonly`. Shouldn't it be the same?

File: src/pyocr/tesseract.py [hunk 2]
```diff
@@ -376,6 +376,40 @@ def image_to_string(image, lang=None, builder=None):
...
+            image.convert("RGB")
+        image.save(os.path.join(tmpdir, "input.bmp"))
+        (status, errors) = run_tesseract("input.bmp", "output", cwd=tmpdir, lang=lang,
+                                         flags=["--psm", "1"], configs=["pdf"])
+        if status:
+            raise TesseractError(status, errors)
+
+        output_file_name = '%s.%s' % (os.path.join(tmpdir, "output"), "pdf")
+        if not os.access(output_file_name, os.F_OK):
+            raise TesseractError(-1, "Unable to find output file %s" % output_file_name)
```
Comment (lubo) [inline src/pyocr/tesseract.py]: Using `%` for string formatting is deprecated in Python 3. You should use [str.format](https://docs.python.org/3/library/stdtypes.html#str.format) instead.

Comment (mskrip) [reply src/pyocr/tesseract.py]: Yes, but to my knowledge  `pyocr` tries to be compatible with Python 2 also

Comment (lubo) [reply src/pyocr/tesseract.py]: AFAIK there's no problem with using `str.format` in Python 2: https://docs.python.org/2/library/stdtypes.html#str.format

File: src/pyocr/tesseract.py (hunk 1 above)
Comment (mskrip) [reply src/pyocr/tesseract.py]: Fixed in a1d6ab5

File: src/pyocr/tesseract.py (hunk 2 above)
Comment (mskrip) [reply src/pyocr/tesseract.py]: Fixed in fb17865

File: src/pyocr/tesseract.py [hunk 3]
```diff
@@ -376,6 +377,57 @@ def image_to_string(image, lang=None, builder=None):
...
+
+        output = output_file if output_file else os.path.join(tmpdir, "output")
+
+        input_path = os.path.join(tmpdir, "input.bmp")
+        image.save(input_path)
+
+        (status, errors) = run_tesseract(input_path, output, lang=lang,
+                                         flags=["--psm", "1", "-c",
+                                                "textonly_pdf={}".format(to)],
+                                         configs=["pdf"])
```
Comment (lubo) [inline src/pyocr/tesseract.py]: Perhaps a better formatting would be:

```python
status, errors = run_tesseract(
    input_path, output, lang=lang,  flags=[
        "--psm", "1", "-c", "textonly_pdf={}".format(to)
    ],
    configs=["pdf"]
)
```

What do you think?

File: src/pyocr/tesseract.py [hunk 4]
```diff
@@ -376,6 +377,57 @@ def image_to_string(image, lang=None, builder=None):
...
+        if status:
+            raise TesseractError(status, errors)
+
+        output_file_name = '{}.{}'.format(output, "pdf")
+
+        if not os.access(output_file_name, os.F_OK):
+            raise TesseractError(-1,
+                                 "Unable to find output file {}".format(output_file_name))
```
Comment (lubo) [inline src/pyocr/tesseract.py]:
```python
raise TesseractError(
    -1, "Unable to find output file {}".format(output_file_name)
)
```

Maybe?

Comment (mskrip) [reply src/pyocr/tesseract.py]: Changed in 731a36d

File: src/pyocr/tesseract.py (hunk 3 above)
Comment (mskrip) [reply src/pyocr/tesseract.py]: Changed in 731a36d

**Pull request merged**